delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/error/TokenError.scala

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.error
+
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.contexts
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.ContextValue
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClientError
+import io.circe.syntax.{EncoderOps, KeyOps}
+import io.circe.{DecodingFailure, Encoder, JsonObject}
+
+sealed abstract class TokenError(reason: String) extends SDKError {
+  override def getMessage: String = reason
+}
+
+object TokenError {
+
+  /**
+    * Signals that an HTTP error occurred when fetching the token
+    */
+  final case class TokenHttpError(cause: HttpClientError)
+      extends TokenError(s"HTTP error when requesting token: ${cause.reason}")
+
+  /**
+    * Signals that the token was missing from the authentication response
+    */
+  final case class TokenNotFoundInResponse(failure: DecodingFailure)
+      extends TokenError(s"Token not found in auth response: ${failure.reason}")
+
+  /**
+    * Signals that the expiry was missing from the authentication response
+    */
+  final case class ExpiryNotFoundInResponse(failure: DecodingFailure)
+      extends TokenError(s"Expiry not found in auth response: ${failure.reason}")
+
+  implicit val identityErrorEncoder: Encoder.AsObject[TokenError] = {
+    Encoder.AsObject.instance[TokenError] {
+      case TokenHttpError(r)           =>
+        JsonObject(keywords.tpe := "TokenHttpError", "reason" := r.reason)
+      case TokenNotFoundInResponse(r)  =>
+        JsonObject(keywords.tpe -> "TokenNotFoundInResponse".asJson, "reason" := r.message)
+      case ExpiryNotFoundInResponse(r) =>
+        JsonObject(keywords.tpe -> "ExpiryNotFoundInResponse".asJson, "reason" := r.message)
+    }
```

Comment:
I tried semi-automatic derivation, but the error messages are bad and I couldn't see the problem

Reply:
What do you mean ?
Also if you only use the message/reason from `DecodingFailure`/  `HttpClientError` because it is what we are interested in here, you should not run into this

Reply:
When using semi-automatic derivation for an Encoder[TokenError], there were errors which were unclear. I tried changing some obvious things, but could not fix the issue. I noticed that we had used this method elsewhere in the codebase, so I did that.